html/rendering/widgets/baseline-alignment-and-overflow.tentative.html

```diff
@@ -0,0 +1,165 @@
+  <tr><td><span>submit</span> <input type=submit value=x></td></tr>
+  <tr><td><span>image</span> <input type=image src="data:,broken" alt="x"></td></tr>
+  <tr><td><span>image-with-src</span> <input type=image src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABgAAAAMCAIAAAD3UuoiAAAAGklEQVQoz2Nk%2BP%2BfgRqAiYFKYNSgUYOGp0EA%2BQMCFrJdTgsAAAAASUVORK5CYII%3D" alt="x"></td></tr>
+  <tr><td><span>reset</span> <input type=reset value=x></td></tr>
+  <tr><td><span>button</span> <input type=button value=x></td></tr>
+</table>
+<h2>tests</h2>
+<!--
+This table gets populated by the script.
+-->
+<table class="test" id="test-table"><tbody></tbody></table>
+<script>
+  "use strict";
+
+  promise_setup(async () => {
+    const templateTable = document.querySelector('#template-table');
+    const testTBody = document.querySelector('#test-table tbody');
+
+    const templateRows = templateTable.querySelectorAll('tr');
+    for (let templateRow of templateRows) {
+      for (var appearanceValue of ["auto", "none"]) {
+        for (var overflowValue of ['visible', 'hidden', 'scroll']) {
+          const clonedRow = templateRow.cloneNode(true);
+          clonedRow.querySelector('input').setAttribute('style', `overflow: ${overflowValue}; appearance: ${appearanceValue};`);
+          testTBody.append(clonedRow);
+        }
+      }
+    }
+
+    // wait for images to load
+    await new Promise(resolve => window.onload = e => resolve());
+    for (const img of document.images) {
+      assert_precondition(img.complete); // either error state or loaded
+    }
+
+    // get layout info from refs
+    const refTextOffsetTop = document.querySelector('.ref-text span').offsetTop;
+    const refCheckboxOffsetTop = document.querySelector('.ref-checkbox span').offsetTop;
+    const refColorOffsetTop = document.querySelector('.ref-color span').offsetTop;
+    const refFileOffsetTop = document.querySelector('.ref-file span').offsetTop;
+    const refImageAltOffsetTop = document.querySelector('.ref-image-alt span').offsetTop;
+    const refImageSrcOffsetTop = document.querySelector('.ref-image-src span').offsetTop;
+
+    function expectedOffsetTop(input) {
+      // TODO(zcorpan) https://github.com/whatwg/html/issues/5065
+      // for now this is intended to match Firefox
+      const style = input.getAttribute('style');
+      const src = input.getAttribute('src');
+      switch (input.type) {
+        case 'file':
+          return refFileOffsetTop;
+        case 'range':
+          return refImageSrcOffsetTop;
+        case 'color':
+          return refColorOffsetTop;
+        case 'checkbox':
+        case 'radio':
+          return (style.includes('appearance: none;')) ? refImageSrcOffsetTop : refCheckboxOffsetTop;
+        case 'file':
+          return refFileOffsetTop;
+        case 'image':
+          return (src === 'data:,broken' && style.includes('overflow: visible;')) ? refImageAltOffsetTop : refImageSrcOffsetTop;
+      }
+      return refTextOffsetTop;
```

Comment:
If you put this in the `default:` of the switch I think it'd be a bit easier to scan.

Reply:
Done